File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& arr) {
        int smallest=arr[0];
        int mx_profit=0;
        for(int i=1;i<arr.size();i++){
            if(arr[i]<smallest)smallest=arr[i];
            else mx_profit=max(mx_profit,arr[i]-smallest);
        }
        return mx_profit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] arr) {
        int mn = arr[0];
        int ans  = 0,l=0,r=1;
        while(r< arr.length){
            if(arr[r]<arr[l]){
                l=r;
            }
            else{
                ans = Math.max(ans,arr[r]-arr[l]);
            }
            r++;
        }
        return ans;
    }
}